### PR TITLE
add saveAs function

### PR DESCRIPTION
**Pull request by hanszhoou**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
     "react-router": "^5.2.0",
     "react-use": "^15.3.3",
     "rimraf": "^3.0.2",
-    "typescript": "^3.9.7"
+    "typescript": "^3.9.7",
+    "web-ext-types": "^3.2.1"
```
Comment (hanszhoou) [inline package.json:23]: declare `browser` type

Comment (septs) [reply package.json:23]: this is peer dep

File: src/saveAs.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { once } from 'lodash-es'
+
+const createDlElement = once((url: string, suggestingFileName: string) => {
+  const dlElem = document.createElement('a')
+  dlElem.setAttribute('href', url)
+  dlElem.setAttribute('download', suggestingFileName)
+  return dlElem
+})
```
Comment (septs) [inline src/saveAs.ts]:
```suggestion
const createDownloadElement = (href: string, fileName = "") => {
  const element = document.createElement('a')
  element.href = url
  element.download = fileName
  return element
}
```

this is wrong usage

```
> wrap = _.once((foo) => foo)
> wrap(1)
< 1
> wrap(2)
< 1
> wrap()
< 1
```

File: src/saveAs.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export function saveAsFile(file: ArrayBuffer, mineType: string, suggestingFileName: string, isAndroid: boolean) {
```
Comment (septs) [inline src/saveAs.ts]:
```suggestion
export function saveAsFile(source: BufferSource, mimeType: string, fileName = "", isAndroid = true) {
```

**Review by septs — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (hanszhoou) [reply package.json:23]: when use peer dep, throw error:

image

**Review by septs — APPROVED** [review]

**Pull request merged**